cpp/proto/xjson_serializer.cc

```diff
@@ -62,16 +62,16 @@ SerializeResult SerializeV1SCTSignatureInput(
   if (res != SerializeResult::OK) {
     return res;
   }
-  TLSSerializer serializer;
-  serializer.WriteUint(ct::V1, Serializer::kVersionLengthInBytes);
-  serializer.WriteUint(ct::CERTIFICATE_TIMESTAMP,
-                       Serializer::kSignatureTypeLengthInBytes);
-  serializer.WriteUint(sct.timestamp(), Serializer::kTimestampLengthInBytes);
-  serializer.WriteUint(ct::X_JSON_ENTRY,
-                       Serializer::kLogEntryTypeLengthInBytes);
-  serializer.WriteVarBytes(json, kMaxJsonLength);
-  serializer.WriteVarBytes(extensions, Serializer::kMaxExtensionsLength);
-  result->assign(serializer.SerializedString());
```

Comment:
You should add a `result->clear();` call just before doing the writing, to maintain the semantics of this function.